src/myloader.c

```diff
@@ -112,6 +112,21 @@ GHashTable * myloader_initialize_hash_of_session_variables(){
   if (commit_count > 1)
     g_hash_table_insert(set_session_hash,g_strdup("AUTOCOMMIT"),g_strdup("0"));
 
+  const int m= get_major();
+  const int s= get_secondary();
+  const int r= get_revision();
+
+  if (m >= 4) {
+    if (m > 4 || s > 0 || r >= 14) {
+      g_hash_table_insert(set_session_hash,g_strdup("UNIQUE_CHECKS"), g_strdup("0"));
+      g_hash_table_insert(set_session_hash,g_strdup("FOREIGN_KEY_CHECKS"), g_strdup("0"));
+    }
+    if (m > 4 || s > 1 || (s == 1 && r >= 1))
+      g_hash_table_insert(set_session_hash,g_strdup("SQL_MODE"), g_strdup("'NO_AUTO_VALUE_ON_ZERO'"));
+    if (m > 4 || s > 1 || (s == 1 && r >= 11))
+      g_hash_table_insert(set_session_hash,g_strdup("SQL_NOTES"), g_strdup("0"));
```

Comment:
that is why we have mydumper.cnf, to avoid this.

Reply:
Configuration file should not be mandatory for correct work of software. This is user unfriendly! Otherwise you should fail it to run if it doesn't find configuration file. What exactly did you want to avoid? There is only 4 variables and the number doesn't grow for long time.

Reply:
I moved this commit to PR #1339 so we could discuss it separately.